src/_pytest/fixtures.py

```diff
@@ -503,15 +540,16 @@ def _get_active_fixturedef(self, argname):
         self._fixture_defs[argname] = fixturedef
         return fixturedef
 
-    def _get_fixturestack(self):
+    def _get_fixturestack(self) -> List["FixtureDef"]:
         current = self
-        values = []
+        values = []  # type: List[FixtureDef]
         while 1:
             fixturedef = getattr(current, "_fixturedef", None)
             if fixturedef is None:
                 values.reverse()
                 return values
             values.append(fixturedef)
+            assert isinstance(current, SubRequest)
```

Comment:
This attribute is only set on `SubRequest`.

Reply:
I don't know, once the assert is in, the implication is reversed -- the access is legal because of the assert, not the other way :)